mkosi.md

```diff
@@ -1329,6 +1335,70 @@ turn-around times for complete image builds are minimal, as only
 changed source files need to be recompiled: an OS image rebuilt will
 be almost as quick to build the source tree only.
 
+# PRESETS
+
+A preset is just like a regular configuration, except that mkosi can
+build multiple presets sequentially. Presets can be used to separate builds into stages,
+to build InitRds, or to build tools used at a later point.
+
+Presets are located as subdirectories in **`mkosi.presets/`**.
+If this path exists, for each preset mkosi will read the global configuration, followed by
+the individual preset configuration. 
+
+mkosi reads and builds each of the presets in alpha-numerical order. Ordering can be enforced with a
+numeric prefix
+Alternatively, the presets to build can be specified as command line arguments with `--preset`.
+
+Later presets can use outputs of earlier presets.
+Output files are named based on the preset name, although without any numeric prefix.
+Paths in the `mkosi.conf` are relative to itself.
+For example, a base preset with `Output=directory` produces a `base` directory in
+the output directory. This can be used in a later preset with `BaseTrees=base`, where
+the `base` output is copied in at the start.
+In conjunction with the `Overlay=` option, a system extension image can be generated.
+Presets are also useful to build custom initrds.
+They can be generated with `Output=cpio` and used for UKI generation in a later
+preset with the `Initrds=` option.
+
+## Examples
+
+This example build a base filesystem that is used in a InitRd and the root filesystem.
+The base filesystem could e.g. contain systemd components that are used in both the InitRd and the root filesystem.
+
+Each preset directory can additionally contain scripts that builds the respective preset.
+A more in-depth example can be found in the main systemd repository: https://github.com/systemd/systemd/tree/main/mkosi.presets
+
+
+```
+# mkosi.presets/00-base/mkosi.conf
+[Output]
+Format=directory
+
+[Content]
+Packages=build-essential
```

Comment:
I think the example needs to specify that this is Debian. Otherwise `build-essential` has no meaning.